slack-archive/src/main/java/juja/microservices/slack/archive/api/SlackApiImpl.java

```diff
@@ -0,0 +1,45 @@
+package juja.microservices.slack.archive.api;
+
+import juja.microservices.slack.archive.exceptions.ArchiveException;
+import juja.microservices.slack.archive.model.dto.ChannelDTO;
+import juja.microservices.slack.archive.model.dto.ChannelSlackResponse;
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.beans.factory.annotation.Value;
+import org.springframework.http.ResponseEntity;
+import org.springframework.stereotype.Repository;
+import org.springframework.web.client.HttpClientErrorException;
+import org.springframework.web.client.RestTemplate;
+
+import java.util.List;
+
+@Repository
+@Slf4j
+public class SlackApiImpl implements SlackApi {
```

Comment:
Maybe is better naming  for interface SlackApiClient?

Reply:
done